mesheryctl/pkg/utils/healthcheck.go

```diff
@@ -161,7 +161,7 @@ func IsMesheryRunning(currPlatform string) (bool, error) {
 			deploymentList, err := deploymentInterface.List(context.TODO(), metav1.ListOptions{})
 
 			if err != nil {
-				return false, err
+				return false, ErrConnectionRefused()
```

Comment:
are you sure this is the only type of error that will be raised ?

Reply:
Since the api is called internally, there is no other error that comes to my mind

Reply:
should i update the error to include part of the original error from the k8 api?

Reply:
If there is no connection refused error but something failed in the server itself for example we need to now the root cause

Reply:
The current state is we get the following error only when the cluster is unreachable
 failed to check Meshery Server status: Get "https://192.168.49.2:8443/apis/apps/v1/namespaces/meshery/deployments": dial tcp 192.168.49.2:8443: connect: no route to host
 When we can call the k8s api but meshery is not running then we already get the following error->
 Meshery Server is not running. Run "mesheryctl system start" to start Meshery.

Reply:
So do we really have to change anything then?

Reply:
Let's say it is up and running, it could also raise another error and the message will not reflect it. I'll take a look to give advise.

@Jougan-0, @MUzairS15 am I right ?

Reply:
Yes @lekaf974, you are correct.

@pranjalg1331 Instead define an error `ErrMesheryHealthCheck(err error)` and pass the error into this Meshkit error, so that we are aware this error occurred while performing health check and also includes the root cause.

Reply:
@MUzairS15 updated